### PR TITLE
Update to Discord Integration 3.0.2

### PR DESCRIPTION
**Pull request by CanadianBaconBoi**
Updated to Discord Integration 3.0.2.
Added explicit Gradle version as ForgeGradle 5.x does not support Gradle 8+.
Improved compatibility of MixinForgeMessageUtils with future versions by changing full method overwrite to an inject with locals.
Removed redundant channel checks as they are already done inside of DiscordIntegration off of the main thread.

### CODE REVIEW

File: src/main/java/lv/id/bonne/vhdiscord/dcintegration/mixin/MixinPlayerVaultInteraction.java [hunk 1]
```diff
@@ -61,21 +59,12 @@ private void broadcastMessageToDiscord(Component p_11265_, ChatType p_11266_, UU
...
             {
-                MessageChannel channel =
-                    Variables.discord_instance.getChannel(Configuration.instance().advanced.serverChannelID);
-
-                if (channel == null)
-                {
-                    // Channel is not setup. Do nothing.
-                    return;
-                }
-
```
Comment (CanadianBaconBoi) [inline src/main/java/lv/id/bonne/vhdiscord/dcintegration/mixin/MixinPlayerVaultInteraction.java:74]: Removed as this check is done off-thread inside of DiscordIntegration.

File: src/main/java/lv/id/bonne/vhdiscord/dcintegration/mixin/MixinMiscUtils.java [hunk 2]
```diff
@@ -35,19 +33,10 @@ public class MixinMiscUtils
...
         {
-            MessageChannel channel =
-                Variables.discord_instance.getChannel(Configuration.instance().advanced.serverChannelID);
-
-            if (channel == null)
-            {
-                // Channel is not setup. Do nothing.
-                return;
-            }
-
```
Comment (CanadianBaconBoi) [inline src/main/java/lv/id/bonne/vhdiscord/dcintegration/mixin/MixinMiscUtils.java:48]: Removed as this check is done off-thread inside of DiscordIntegration.

File: src/main/java/lv/id/bonne/vhdiscord/dcintegration/mixin/MixinForgeMessageUtils.java [hunk 3]
```diff
@@ -10,246 +10,53 @@
...
+    @Inject(method = "genItemStackEmbedIfAvailable",
+            at = @At(value = "INVOKE_ASSIGN", target = "Lnet/minecraft/world/item/ItemStack;getOrCreateTag()Lnet/minecraft/nbt/CompoundTag;"),
+            cancellable = true,
+            locals = LocalCapture.CAPTURE_FAILSOFT)
+    private static void genItemStackEmbedIfAvailable(Component component, CallbackInfoReturnable<MessageEmbed> cir,
+        JsonObject json, JsonArray args, Iterator var3,
+        JsonElement el, JsonObject arg1, JsonObject hoverEvent,
+        JsonObject item, ItemStack is, CompoundTag itemTag) {
+        if (item.get("id").getAsString().startsWith("the_vault")) {
+            cir.setReturnValue(MixinForgeMessageUtils.vaultHuntersDiscordIntegrator$craftVaultHuntersItemMessage(item, is, itemTag));
```
Comment (CanadianBaconBoi) [inline src/main/java/lv/id/bonne/vhdiscord/dcintegration/mixin/MixinForgeMessageUtils.java:47]: An Inject mixin is more appropriate as it ensures new features in DiscordIntegration are preserved.

**Comment by CanadianBaconBoi** [comment]
Do note this version is no longer compatible with Discord Integration 2.x.

File: src/main/java/lv/id/bonne/vhdiscord/dcintegration/mixin/MixinForgeMessageUtils.java (hunk 3 above)
Comment (BONNe) [reply src/main/java/lv/id/bonne/vhdiscord/dcintegration/mixin/MixinForgeMessageUtils.java:47]: I did overwrite because I disliked spaghetti code in DC.

Comment (KoromaruKoruko) [reply src/main/java/lv/id/bonne/vhdiscord/dcintegration/mixin/MixinForgeMessageUtils.java:47]: that is the dumbest reason to completely rewrite a method via a mixin.

that just breaks the ability for any other mod to mixin to that method. for 0 objective reason at all.

Comment (BONNe) [reply src/main/java/lv/id/bonne/vhdiscord/dcintegration/mixin/MixinForgeMessageUtils.java:47]: I do not think any other mod is injecting anything in that method :) In general, I agree.

Comment (CanadianBaconBoi) [reply src/main/java/lv/id/bonne/vhdiscord/dcintegration/mixin/MixinForgeMessageUtils.java:47]: If you believe the code in another mod is spaghetti, I suggest creating a PR in that mod's repository.
Completely overwriting a method using a mixin in order to refactor is not appropriate in most circumstances.

**Comment by BONNe** [comment]
Thanks for the Pull Request

**Pull request merged**